android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Router.kt

```diff
@@ -206,11 +206,11 @@ protected constructor(
   public open fun dispatchDetach() {
     checkForMainThread()
 
-    interactorGeneric.dispatchDetach()
     willDetach()
```

Comment:
Should the router `willDetach` come *after* interactor detach? IMO this should be the last line. What issues did you encounter with this (if any)?

Reply:
Upon second though, maybe calling willDetach before detaching children and self makes some sense after all. "Will" implies some kind of preparation for something that is about to happen.

Reply:
What I had noticed was that some existing RIBs in Uber's app are overriding the `willDetach()` (and sometimes even call `detachChild()` themselves e.g. BikeHomeRouter) and/or do other sorts of clean up stuff that needs to happen before `detachChild()` is called by Router.

I think those seemingly ani-pattern-ish usage may be symptoms of the fact that we are detaching parent before child.

Reply:
I think you are right.

I've been thinking about this, there's nothing super wrong with calling willDetach before everything else. It can be understood as a signal that router X is **starting** its detach process, which now will _begin_ with detaching children (instead with detaching self). That makes sense.